rust/src/client/request.rs

```diff
@@ -1,7 +1,35 @@
+//! Provides functionality to enable clients to communicate with a XayNet
+//! service over HTTP.
+//!
+//! This includes a [`Proxy`] which a [`Client`] can use to communicate with the
+//! service. To summarise, the proxy:
+//!
+//! * Wraps either a [`Handle`] (for local comms) or a [`ClientReq`] (for remote
+//!   comms over HTTP).
+//! * In the latter case, deals with logging and wrapping of network errors.
+//! * Deals with deserialization
+//!
+//! [`ClientReq`] is responsible for building the HTTP request and extracting
+//! the response body. As an example:
+//!
+//! `async fn get_sums(&self) -> Result<Option<bytes::Bytes>, reqwest::Error>`
```

Comment:
i've seen that you removed the code block because of running doc tests. if you want to keep the block, you can annotate it with `no_rust` which stops rustdoc from running this code. like
```
//! ```no_rust
//! // code
//! ```
```